qemu/tests/cpu_topology_test.py

```diff
@@ -0,0 +1,75 @@
+import re
+import random
+
+from avocado.utils import cpu
+
+from virttest import error_context
+from virttest import env_process
+
+from provider.cpu_utils import check_guest_cpu_topology
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check guest gets correct vcpu num, cpu cores, processors, sockets, siblings
+
+    1) Boot guest with options: -smp n,cores=x,threads=y,sockets=z...
+    2) Check cpu topology
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def check(p_name, exp, check_cmd):
+        """
+        Check the cpu property inside guest
+        :param p_name: Property name
+        :param exp: The expect value
+        :param check_cmd: The param to get check command
+        """
+        res = session.cmd_output(params[check_cmd]).strip()
+        if int(res) != int(exp):
+            test.fail('The vcpu %s number inside guest is %s,'
+                      ' while it is set to %s' % (p_name, res, exp))
+
+    vm_name = params['main_vm']
+    os_type = params['os_type']
+    vcpu_threads_list = [1, 2]
+    if params['machine_type'] == 'pseries':
+        vcpu_threads_list = [1, 2, 4, 8]
+    params['vcpu_cores'] = vcpu_cores = random.randint(1, 10)
+    host_cpu = cpu.online_count()
```

Comment:
Can we get the vcpu_cores according to the host_cpu to avoid CPU overcommit?

Reply:
@nanliu-r Sure, what do you expect? Same as host cores?

Reply:
> @nanliu-r Sure, what do you expect? Same as host cores?

Sounds good for me.

Reply:
How about half of host_cpu considering of multiple threads?

Reply:
@nanliu-r What if host cpu core is 2? Will that be too small? Let's try with =host cpu and see if it works well first, what do you think?

Reply:
@nanliu-r After a second thought, I don't think that's a good idea, the host cpu might be very large, also the calculation of vcpu_sockets will be meaningless if so. Let me know your thought, thanks.

Reply:
I update `vcpu_cores` from `andom.randint(1, 10)` to `random.randint(1, 6)`.

Reply:
> @nanliu-r After a second thought, I don't think that's a good idea, the host cpu might be very large, also the calculation of vcpu_sockets will be meaningless if so. Let me know your thought, thanks.

I mean we can consider two situations:
1. when host_cpu >= 12 (since you changed cores in range(1, 6)) we can keep the current configuration. And even host_cpu is a large number we won't get a large SMP number.
2. when host_cpu <12 we can change the cores in range(1, host_cpu/2). This just is to avoid CPU overcommit.

@vivianQizhu  How do you think?

Reply:
@nanliu-r Update to `vcpu_cores = random.randint(1, max(6, host_cpu//2))`. Please help review, thanks.

Reply:
Shouldn't it be `vcpu_cores = random.randint(1, min(6, host_cpu//2))` ?

Reply:
@nanliu-r No, you said, when host_cpu<12(hostcpu//2 < 6), it should be 6, when host_cpu >=12(host_cpu//2 > 6), it should be host_cpu//2, so it should be max(6, host_cpu//2).

Reply:
@vivianQizhu `vcpu_cores = random.randint(1, min(10, host_cpu//2))` This is what I said at first.

Reply:
@nanliu-r Okay, updated.